src/khoj/processor/conversation/gpt4all/utils.py

```diff
@@ -35,7 +37,14 @@ def download_model(model_name):
                 for chunk in r.iter_content(chunk_size=8192):
                     f.write(chunk)
                     progress_bar.update(len(chunk))
+
+        logger.debug(f"Successfully downloaded model {model_name} from {url} to {tmp_filename}")
+        # Move the tmp file to the actual file
+        os.rename(tmp_filename, filename)
         return GPT4All(model_name)
     except Exception as e:
-        logger.error(f"Failed to download model {model_name} from {url} to {filename}. Error: {e}")
+        logger.error(f"Failed to download model {model_name} from {url} to {tmp_filename}. Error: {e}")
```

Comment:
Given the error message is user facing, should it mention the file path the model is to be stored at, instead of it's temp path?

```suggestion
        logger.error(f"Failed to download model {model_name} from {url} to {filename}. Error: {e}")
```

Reply:
I think tmp path is more relevant here, in case the user needs to clean up any download artifacts.

Reply:
Oh, don't we already do that cleanup of the temp filename for them below?

Reply:
True, let's update it.